client/dashboard/sites/emails/index.tsx

```diff
@@ -1,19 +1,205 @@
 import PageLayout from '../../components/page-layout';
+import GoogleLogo from './resources/google-logo';
+import type { View } from '@wordpress/dataviews';
+import type React from 'react';
+
+import './styles.scss';
+
+const fields = [
+	{
+		id: 'emailAddress',
+		label: __( 'Email Address' ),
+		enableGlobalSearch: true,
+		render: ( { item }: { item: Email } ) => {
+			let iconEl = <Icon icon={ wordpress } size={ 28 } className="professional-email-icon" />;
+			if ( item.type === 'forwarding' ) {
+				iconEl = <Icon icon={ next } size={ 28 } className="email-forwarder-icon" />;
+			}
+
+			if ( item.type === 'mailbox' && item.provider === 'google_workspace' ) {
+				iconEl = <GoogleLogo size={ 24 } className="google-workspace-email-icon" />;
+			}
+
+			return (
+				<HStack spacing={ 4 } justify="flex-start">
+					<div className="email-icon-wrapper">{ iconEl }</div>
+					{ item.type === 'mailbox' ? (
+						<ExternalLink href={ `https://mail.${ item.domainName }` }>
+							{ item.emailAddress }
+						</ExternalLink>
+					) : (
+						<VStack justify="flex-start">
+							<span>{ item.emailAddress }</span>
+							<span className="text-muted">
+								{ sprintf(
+									/* translators: %s is the email messages will be forwarded to. */
+									__( 'Forwards to %s' ),
+									item.forwardingTo
+								) }
+							</span>
+						</VStack>
+					) }
+				</HStack>
+			);
+		},
+		getValue: ( { item }: { item: Email } ) => item.emailAddress,
+	},
+	{
+		id: 'domainName',
+		label: __( 'Domain' ),
+		getValue: ( { item }: { item: Email } ) => item.domainName,
+	},
+	{
+		id: 'type',
+		label: __( 'Type' ),
+		render: ( { item }: { item: Email } ) =>
+			item.type === 'mailbox' ? __( 'Mailbox' ) : __( 'Forwarder' ),
+		getValue: ( { item }: { item: Email } ) => item.type,
+		elements: [
+			{ value: 'mailbox', label: __( 'Mailbox' ) },
+			{ value: 'forwarding', label: __( 'Forwarder' ) },
+		],
+	},
+	{
+		id: 'status',
+		label: __( 'Status' ),
+		render: ( { item }: { item: Email } ) => {
+			if ( item.status === 'active' ) {
+				return __( 'Active' );
+			}
+			if ( item.status === 'pending' ) {
+				return __( 'Pending verification' );
+			}
+			if ( item.status === 'suspended' ) {
+				return __( 'Expired' );
+			}
+			return item.status;
+		},
+		getValue: ( { item }: { item: Email } ) => item.status,
+		// map to display values for filtering UI
+		elements: [
+			{ value: 'active', label: __( 'Active' ) },
+			{ value: 'pending', label: __( 'Pending verification' ) },
+			{ value: 'suspended', label: __( 'Expired' ) },
+		],
+	},
+];
 
 function SiteEmails() {
+	const navigate = useNavigate();
 	const { siteSlug } = siteRoute.useParams();
-	const { data: site } = useQuery( siteBySlugQuery( siteSlug ) );
+	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
+
+	// Wait for site to load to know ID
+	const { data: domains, isLoading: isDomainsLoading } = useQuery(
+		site && site.ID
```

Comment:
`site` and `site.ID` is always defined by now. We don't need to wait and we don't need this conditionals.